drivers/sensors/mpu60x0.c

```diff
@@ -639,6 +664,85 @@ static inline int __mpu_write_config(FAR struct mpu_dev_s *dev,
   return __mpu_write_reg(dev, CONFIG, &val, sizeof(val));
 }
 
+/* Sets the SMPLRT_DIV that controls the sample rate. */
+
+static inline int __mpu_set_sample_rate_divider(FAR struct mpu_dev_s *dev,
+                                                uint8_t val)
+{
+  return __mpu_write_reg(dev, SMPLRT_DIV, &val, sizeof(val));
+}
+
+/* Reads current sample rate. Value is updated to mpu_dev_s->sample_rate. */
+
+static inline int __mpu_read_sample_rate(FAR struct mpu_dev_s *dev)
+{
+  int ret;
+  float gyro_output_rate = 1000.0f;
+
+  ret = __mpu_read_reg(dev, SMPLRT_DIV, &dev->smplrt_div,
+                       sizeof(dev->smplrt_div));
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  ret = __mpu_read_reg(dev, CONFIG, &dev->dlpf_config,
+                       sizeof(dev->dlpf_config));
+  if (ret < 0)
+    {
+      return ret;
+    }
+
+  dev->dlpf_config = TO_BITFIELD(CONFIG__DLPF_CFG, dev->dlpf_config);
+
+  /* This condition verifies if DLPF is disabled */
+
+  if ((dev->dlpf_config == 0) || (dev->dlpf_config == 7))
+    {
+      gyro_output_rate = 8000.0f;
+    }
+
+  dev->sample_rate = gyro_output_rate / (float)(1 + dev->smplrt_div);
+
+  return OK;
+}
+
+/* Read the number of bytes currently in FIFO buffer. */
+
+static inline int __mpu_read_fifo_count(FAR struct mpu_dev_s *dev,
+                                        uint16_t *buf)
+{
+  int ret;
+  uint8_t fifo_counter[2];
+  ret = __mpu_read_reg(dev, FIFO_COUNTH, fifo_counter, sizeof(fifo_counter));
+  *buf = (fifo_counter[0] << 8) | fifo_counter[1];
```

Comment:
should we modify `*buf` content if `__mpu_read_reg` fails? I'm fine with it, just thinking from the overall perspective

Reply:
Yes, I think that will add a great safety net.